crates/bevy_transform/src/components/global_transform.rs

```diff
@@ -211,6 +211,32 @@ impl GlobalTransform {
         self.0.translation
     }
 
+    /// Get the rotation as a [`Quat`].
+    ///
+    /// The transform is expected to be non-degenerate and without shearing, or the output will be invalid.
+    ///
+    /// # Warning
+    ///
+    /// This is calculated using `to_scale_rotation_translation`, meaning that you
+    /// should probably use it directly if you also need translation or scale.
+    #[inline]
+    pub fn rotation(&self) -> Quat {
+        self.to_scale_rotation_translation().1
+    }
+
+    /// Get the scale as a [`Vec3`].
+    ///
+    /// The transform is expected to be non-degenerate and without shearing, or the output will be invalid.
+    #[inline]
+    pub fn scale(&self) -> Vec3 {
+        let det = self.0.matrix3.determinant();
+        Vec3::new(
+            self.0.matrix3.x_axis.length() * det.signum(),
```

Comment:
Can you leave a comment here explaining why only the x value is multiplied by the sign of the determinant?

Reply:
I mirrored it from the glam code, without knowing much of the math behind. 

A simpler implementation would be to just `to_scale_rotation_translation().0` but there would be a slight performance penalty. Whatever you think is best.

https://docs.rs/glam/latest/src/glam/f32/affine3a.rs.html#285-309

Reply:
You could put that link as a normal comment there

Reply:
Okay, did that